File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/model/IDataRestServer.java
package ru.kmorozov.library.data.model;

import ru.kmorozov.library.data.model.dto.*;
import ru.kmorozov.library.data.model.dto.ItemDTO.ItemType;

import java.util.List;

/**
 * Created by sbt-morozov-kv on 17.01.2017.
 */
public interface IDataRestServer {

    UserDTO login(String login);

    List<StorageDTO> getStoragesByParentId(String storageId);

    List<BookDTO> getBooksByStorageId(String storageId);

    List<ItemDTO> getItemsByStorageId(String storageId);

    ItemDTO getItem(String itemId, ItemType itemType, boolean refresh);

    void updateLibrary(String state);

    BookDTO downloadBook(String bookId);

    List<DuplicatedBookDTO> findDuplicates();

    void synchronizeDb();
}

File: ru.kmorozov.library.data.model/src/main/java/ru/kmorozov/library/data/model/IDataRestServer.kt
package ru.kmorozov.library.data.model

import ru.kmorozov.library.data.model.dto.*
import ru.kmorozov.library.data.model.dto.ItemDTO.ItemType

/**
 * Created by sbt-morozov-kv on 17.01.2017.
 */
interface IDataRestServer {

    fun login(login: String): UserDTO

    fun getStoragesByParentId(storageId: String): List<StorageDTO>

    fun getBooksByStorageId(storageId: String): List<BookDTO>

    fun getItemsByStorageId(storageId: String): List<ItemDTO>

    fun getItem(itemId: String, itemType: ItemType, refresh: Boolean): ItemDTO?

    fun updateLibrary(state: String)

    fun downloadBook(bookId: String): BookDTO

    fun findDuplicates(): List<DuplicatedBookDTO>

    fun synchronizeDb()
}
